vision/standard_object/odlc_image_processing.py

```diff
@@ -0,0 +1,67 @@
+"""
+Preprocesses images obtained during airdrop area flyover for
+use in contour detection for the standard objects.
+"""
+
+import cv2
+import numpy as np
+
+from vision.common.constants import Image, Kernel
+
+
+def preprocess_std_odlc(image: Image, thresh_min: int, thresh_max: int) -> Image:
+    """
+    Preprocesses image for use in detecting the contours of standard objects.
+
+    Parameters
+    ----------
+    image : Image
+        image from airdrop area flyover before any processing has occured
+    thresh_min: int
+        The minimum threshold input for the Canny edge detection
+    thresh_max: int
+        The maximum threshold input for the Canny edge detection
+
+    Returns
+    -------
+    edges : Image
+        the image after preprocessing for use in contour detection/processing
+    """
+    grayscaled: Image = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)  # Convert to graycsale
+
+    blurred: Image = cv2.GaussianBlur(grayscaled, ksize=(3, 3), sigmaX=1.5)  # Blur the image
+
+    edges: Image = cv2.Canny(image=blurred, threshold1=thresh_min, threshold2=thresh_max)
+
+    # Create the kernel for the dilation
+    kernel: Kernel = np.ones((3, 3), np.uint8)
+
+    dilated: Image = cv2.dilate(edges, kernel, iterations=1)
```

Comment:
Also, does the dilation have a negative impact on the clarity of the shape?

Reply:
The short answer is no, a small dilation like this shouldn't significantly impact the appearance of the shape - unless there are a bunch of edges around it like in the image you sent earlier.

I did have some success getting rid of most of these extra edges by tweaking the parameters, so hopefully the iterative process I am planning on doing in the pipeline will result in one set of parameters working really well. In that case, a dilation should be helpful.